src/Groups/GAPGroups.jl

```diff
@@ -0,0 +1,617 @@
+
+
+# TODO: as soon as GAP packages like `polycyclic` or `rcwa` are loaded,
+# the custom group types and isos they define should be added to the arrays
+# _gap_group_types resp. _iso_function
+
+
+function group_element(G::T, x::GapObj) where T <: Group
+  return GAPGroupElem{T}(G, x)
+end
+
+function elements(G::T) where T <: Group
+  els = GAP.gap_to_julia(GAP.Globals.Elements(G.X))
+  elems = Vector{elem_type(G)}(undef, length(els))
+  i = 1
+  for x in els
+    elems[i] = group_element(G, x)
+    i += 1
+  end
+  return elems
+end
+
+# to be fixed later
+Base.hash(x::GAPGroupElem) = 0
+
+function parent(x::GAPGroupElem)
+  return x.parent
+end
+
+function Base.isfinite(G::PermGroup)
+  return true
+end
+
+function Base.isfinite(G::PcGroup)
+  return true
+end
+
+function Base.isfinite(G::Group)
+  return GAP.Globals.IsFinite(G.X)
+end
+
+function degree(x::PermGroup)
+   return x.deg
+end
+
+function order(x::Group)
+   return GAP.gap_to_julia(GAP.Globals.Size(x.X))
+end
+
+function order(x::GAPGroupElem)
+   return GAP.gap_to_julia(GAP.Globals.Order(x.X))
+end
+
+function order(::Type{T}, x::Union{GAPGroupElem, Group}) where T<:Number
+   return T(order(x))
+end
+
+Base.exponent(x::Group) = GAP.Globals.Exponent(x.X)
+
+Base.length(x::Group) = order(x)
+
+function Base.rand(x::Group)
+   s = GAP.Globals.Random(x.X)
+   return group_element(x, s)
+end
+
+# one of the following should be non-parametric
+rand_pseudo(G::Group) = Base.rand(G)
+
+
+function _maxgroup(x::T, y::T) where T <: Group
+   if x == y
+     return x
+   elseif GAP.Globals.IsSubset(x.X, y.X)
+     return x
+   elseif GAP.Globals.IsSubset(y.X, x.X)
+     return y
+   else
+     error("Not yet implemented")
+   end
+end
+
+#We need a lattice of groups to implement this properly
+function _prod(x::T, y::T) where T <: GAPGroupElem
+  G = _maxgroup(parent(x), parent(y))
+  return group_element(G, x.X*y.X)
+end
+
+Base.:*(x::GAPGroupElem, y::GAPGroupElem) = _prod(x, y)
+
+function ==(x::Group, y::Group)
+   return x.X == y.X
+end
+
+function ==(x::T, y::T) where T <: GAPGroupElem
+   return x.X == y.X
+end
+
+"""
+    one(G::Group) -> x::GAPGroupElem{typeof(G)}
+Returns the one element of the group G.
+"""
+Base.one(x::Group) = group_element(x, GAP.Globals.Identity(x.X))
+
+"""
+    one(x::GAPGroupElement{T}) -> x::GAPGroupElem{T}
+Returns the one element of the parent of x.
+"""
+Base.one(x::GAPGroupElem) = one(parent(x))
+one!(x::GAPGroupElem) = one(parent(x))
+
+Base.show(io::IO, x::GAPGroupElem) =  print(io, GAP.gap_to_julia(GAP.Globals.StringView(x.X)))
+Base.show(io::IO, x::Group) = print(io, GAP.gap_to_julia(GAP.Globals.StringView(x.X)))
```

Comment:
show for general `Group`?

Reply:
No, this is just for `Oscar.Group`, a subtype of `AbstractAlgebra.Group`. That said, I think we should rename it (back) to `GAPGroup` to avoid confusion!

Reply:
I was confused because there is `GAPGroupElem`, either both should have `GAP` prefix or none ;)

Reply:
Well, we *had* to rename `GroupElem` to `GAPGroupElem` for technical reasons; those reasons don't exist for `Oscar.Group`, which is why it wasn't renamed. But I agree with you, they should match (see also my comment concerning this elsewhere)

Reply:
Sorry, I don't understand the question.

Reply:
@GDeFranceschi here is my interpretation of the question: the method under discussion here, `Base.show(io::IO, x::Group)`, looks as if it was applied to *arbitrary* group objects (even those which are not defined in terms of GAP objects), which of course would be wrong, as for such group objects it'd produce an error as soon. The confusion is there because there are two *different* types named `Group`, as explained above. My suggestion is to resolve this (at least for now) by renaming "our" `Group` back to `GAPGroup`. This name normally won't be visible to users anyway (they'll see `PermGroup` or `PcGroup` etc. instead). 

This can essentially be done with a big search & replace, except of course for the places where we *actually* mean `AbstractAlgebra.Group` (e.g. in the definition of "our" type)

Reply:
Ah, ok. So, do I rename the type `Group` as `GAPGroup` in all the codes of Groups.jl ?